mercantile/__init__.py

```diff
@@ -761,3 +764,60 @@ def feature(
         feat["id"] = fid
 
     return feat
+
+
+def x_minmax(zoom):
+    """Returns the min and max x tile coordinates for a specific zoom level
+
+    Parameters
+    ----------
+    zoom : int
+        The web mercator zoom level.
+
+    Returns
+    -------
+    Tuple of (min, max)
+
+    """
+
+    return (0, 2 ** zoom - 1)
+
+
+def x_minmax(zoom):
+    """Returns the min and max x tile coordinates for a specific zoom level
+
+    Parameters
+    ----------
+    zoom : int
+        The web mercator zoom level.
+
+    Returns
+    -------
+    Tuple of (min, max) integers
+
+    """
+
+    if zoom != int(zoom) or int(zoom) < 0:
+        raise InvalidZoomError("zoom must be a positive integer")
```

Comment:
This validation seems to be the essence of the function, right? Just using `(0, 2 ** z - 1)` is fine in applications except when you get a zoom < 0 for some reason.